### DISCUSSION TITLE
unable to get username when using StreamingClient in v2/4.9.0

### DISCUSSION THREAD

**Discussion by scottydelta** [tweepy/tweepy#1885] [Questions] [answered]
I am trying to get the username of the author of the original tweet when fetching tweets via StreamingClient in v2 of twitter API. But I dont see the username field when I try to print `tweet.data`

Here is the code and the response I am getting:

```
In [8]: import tweepy


In [9]: class IDPrinter(tweepy.StreamingClient):
   ...:
   ...:     def on_tweet(self, tweet):
   ...:         print(tweet.data)
   ...:


In [10]: printer = IDPrinter(BEARER_TOKEN, wait_on_rate_limit=True)

In [11]: printer.filter(tweet_fields=["created_at","author_id"],user_fields=["username"],expansions=["author_id"])

> {'author_id': '581464808', 'created_at': '2022-05-08T20:24:30.000Z', 'id': '1523398458421391360', 'text': 'test'}

```

I am able to get the username when requesting the data directly via Curl:
```
curl 'https://api.twitter.com/2/tweets/search/stream?tweet.fields=created_at,author_id&user.fields=username&expansions=author_id' -H "Authorization: Bearer BEARER_TOKEN"

> {"data":{"author_id":"581464808","created_at":"2022-05-08T20:11:31.000Z","id":"1523395190584610816","text":"test"},"includes":{"users":[{"id":"581464808","name":"Some Userj","username":"someuser"}]},"matching_rules":[{"id":"1523394252348309504","tag":""}]}
```

How can I get the username of the user?

`tweepy version: 4.9.0`
`twitter API version: v2`

**Comment by Harmon758** [comment, marked as answer]
You'll need to use [`StreamingClient.on_includes`](https://tweepy.readthedocs.io/en/v4.9.0/streamingclient.html#tweepy.StreamingClient.on_includes) or [`StreamingClient.on_response`](https://tweepy.readthedocs.io/en/v4.9.0/streamingclient.html#tweepy.StreamingClient.on_response).

`StreamingClient.on_tweet` only gets passed the Tweet in the `data` field, and as you can see, the User object is in a separate `includes` field.

See [Tweepy's documentation on expansions](https://tweepy.readthedocs.io/en/v4.9.0/client.html#expansions-and-fields-parameters) for more information.
